ql/src/java/org/apache/hadoop/hive/ql/ddl/table/info/desc/formatter/TextDescTableFormatter.java

```diff
@@ -0,0 +1,575 @@
+        }
+        formatOutput("Skewed Value to Truncated Path:", truncatedSkewedColMap.toString(), tableInfo);
+      }
+    }
+
+    if (storageDesc.getSerdeInfo().getParametersSize() > 0) {
+      tableInfo.append("Storage Desc Params:" + LINE_DELIM);
+      displayAllParameters(storageDesc.getSerdeInfo().getParameters(), tableInfo);
+    }
+  }
+
+  private void getTableMetaDataInformation(StringBuilder tableInfo, Table table, boolean isOutputPadded) {
+    formatOutput("Database:", table.getDbName(), tableInfo);
+    formatOutput("OwnerType:", (table.getOwnerType() != null) ? table.getOwnerType().name() : "null", tableInfo);
+    formatOutput("Owner:", table.getOwner(), tableInfo);
+    formatOutput("CreateTime:", formatDate(table.getTTable().getCreateTime()), tableInfo);
+    formatOutput("LastAccessTime:", formatDate(table.getTTable().getLastAccessTime()), tableInfo);
+    formatOutput("Retention:", Integer.toString(table.getRetention()), tableInfo);
+    
+    if (!table.isView()) {
+      formatOutput("Location:", table.getDataLocation().toString(), tableInfo);
+    }
+    formatOutput("Table Type:", table.getTableType().name(), tableInfo);
+
+    if (table.getParameters().size() > 0) {
+      tableInfo.append("Table Parameters:" + LINE_DELIM);
+      displayAllParameters(table.getParameters(), tableInfo, false, isOutputPadded);
+    }
+  }
+
+  private void getPartitionMetaDataInformation(StringBuilder tableInfo, Partition partition) {
+    formatOutput("Partition Value:", partition.getValues().toString(), tableInfo);
+    formatOutput("Database:", partition.getTPartition().getDbName(), tableInfo);
+    formatOutput("Table:", partition.getTable().getTableName(), tableInfo);
+    formatOutput("CreateTime:", formatDate(partition.getTPartition().getCreateTime()), tableInfo);
+    formatOutput("LastAccessTime:", formatDate(partition.getTPartition().getLastAccessTime()), tableInfo);
+    formatOutput("Location:", partition.getLocation(), tableInfo);
+
+    if (partition.getTPartition().getParameters().size() > 0) {
+      tableInfo.append("Partition Parameters:" + LINE_DELIM);
+      displayAllParameters(partition.getTPartition().getParameters(), tableInfo);
+    }
+  }
+
+  private class VectorComparator<T extends Comparable<T>>  implements Comparator<List<T>>{
+    @Override
+    public int compare(List<T> listA, List<T> listB) {
+      for (int i = 0; i < listA.size() && i < listB.size(); i++) {
+        T valA = listA.get(i);
+        T valB = listB.get(i);
+        if (valA != null) {
+          int ret = valA.compareTo(valB);
+          if (ret != 0) {
+            return ret;
+          }
+        } else {
+          if (valB != null) {
+            return -1;
+          }
+        }
+      }
+      return Integer.compare(listA.size(), listB.size());
+    }
+  }
+
+  private <T extends Comparable<T>> List<T> sortList(List<T> list){
```

Comment:
Should be able to replace these methods easily with Lambdas...

```list.stream().sorted(...).collect(Collectors.toList());```

Reply:
Fixed, nice catch!